api/types/header/convert/v1/header.go

```diff
@@ -67,11 +65,19 @@ func FromMetadataProto(msg *headerv1.Metadata) header.Metadata {
 
 // ToMetadataProto converts an internal metadata object into a v1 metadata protobuf message.
 func ToMetadataProto(metadata header.Metadata) *headerv1.Metadata {
+	// We map the zero go time to the zero protobuf time (nil).
+	var expires *timestamppb.Timestamp
+	if metadata.Expires.IsZero() {
+		expires = nil
+	} else {
+		expires = timestamppb.New(metadata.Expires)
+	}
+
 	return &headerv1.Metadata{
 		Name:        metadata.Name,
 		Description: metadata.Description,
 		Labels:      metadata.Labels,
-		Expires:     timestamppb.New(metadata.Expires),
+		Expires:     expires,
```

Comment:
Instead of `nil` can we use the default value here? I'm pretty sure that for most people the first instinct will be to call `IsZero()` instead of `!= nil` which will panic.

Reply:
No, this is the point of the change. When Terraform starts with a nil pointer (because no expiration was set), we convert to `api/types` and convert back to protobuf, Terraform expects a nil pointer, not a default value like 0001-01-01 or epoch. If it receives a different value it will stop execution and throw errors.